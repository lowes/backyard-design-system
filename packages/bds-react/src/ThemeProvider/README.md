# @lowes-tech/bds-react/ThemeProvider

## Namespaces

- [ThemeProvider](modules/ThemeProvider.md)

## Interfaces

- [BackyardTheme](interfaces/BackyardTheme.md)
- [ThemeProviderProps](interfaces/ThemeProviderProps.md)

## References

### default

Renames and re-exports [ThemeProvider](README.md#themeprovider)

## Type aliases

### BaseTheme

Ƭ **BaseTheme**: typeof `Theme`

## Variables

### ThemeConsumer

• **ThemeConsumer**: `React.Consumer`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeConsumer`

ThemeConsumer same as 'styled-components'

___

### ThemeContext

• **ThemeContext**: `React.Context`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeContext`

ThemeContext same as 'styled-components'

___

### ThemeProvider

• **ThemeProvider**: `BDSFunctional`<[`ThemeProviderProps`](interfaces/ThemeProviderProps.md)\>

Backyard React Theme Provider

 <ThemeProvider theme="light" font="fellix" someOtherContext="value">
     <Grid.Container>
         ...
             <Button />
         ...
     </Grid.Container>
 </ThemeProvider>

Custom Theme Provider wrapper to provide extra context and overrides for Backyard Themes.

## Functions

### dropShadow

▸ `Const` **dropShadow**(`shadow`, `defaultShadow?`): `string`

Converts box-shadow to drop-shadow, if necessary.

Use for `filter` css instead of `box-shadow`.

**`example`**
```
dropShadow(theme.shadows.shadow_04)
 => 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.2)) 
         drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.14)) 
         drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.12))'
```

#### Parameters

| Name | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| `shadow` | `string` | `undefined` | box-shadow |
| `defaultShadow` | `string` | `'none'` | default shadow if none given |

#### Returns

`string`

___

### getShape

▸ `Const` **getShape**(`shapeProp`, `shapeContext`): ``"rounded"`` \| ``"squared"``

Get the shape or shape context in priority

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shapeProp` | `ExtendedShapes` | shape value from props |
| `shapeContext` | ``"rounded"`` \| ``"squared"`` | shape value from context |

#### Returns

``"rounded"`` \| ``"squared"``

___

### splitShadows

▸ `Const` **splitShadows**(`shadows`, `defaultShadow`): `string`[]

Split the shadows in a given string

**`example`**
```
splitShadows('0px 2px 1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)')
 => [
     '0px 2px 1px rgba(0, 0, 0, 0.2)',
     '0px 1px 1px rgba(0, 0, 0, 0.14)',
     '0px 1px 3px rgba(0, 0, 0, 0.12)'
 ]
```

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shadows` | `string` | shadow string from backyard |
| `defaultShadow` | `string` | default shadow if none given |

#### Returns

`string`[]

___

### useBackyardTheme

▸ `Const` **useBackyardTheme**(`props?`): [`BackyardTheme`](interfaces/BackyardTheme.md)

#### Parameters

| Name | Type |
| :------ | :------ |
| `props` | `UseBackyardThemeProps` |

#### Returns

[`BackyardTheme`](interfaces/BackyardTheme.md)

___

### validateBackyardTheme

▸ `Const` **validateBackyardTheme**(`theme`, `name`): `boolean`

#### Parameters

| Name | Type |
| :------ | :------ |
| `theme` | [`BackyardTheme`](interfaces/BackyardTheme.md) |
| `name` | `string` |

#### Returns

`boolean`
