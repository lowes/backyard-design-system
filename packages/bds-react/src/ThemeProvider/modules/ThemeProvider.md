# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
