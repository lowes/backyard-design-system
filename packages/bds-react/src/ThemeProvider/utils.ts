/* eslint-disable import/prefer-default-export */
import type { ValueOf } from '../utils/typings'
import type { BackyardTheme } from './ThemeProvider'

type ExtendedShapes = BackyardTheme['context']['shape'] | 'circle' | 'custom'

/**
 * Get the shape or shape context in priority
 *
 * @param shapeProp - shape value from props
 * @param shapeContext - shape value from context
 */
export const getShape = (
    shapeProp: ExtendedShapes,
    shapeContext: BackyardTheme['context']['shape'],
) => (shapeProp || shapeContext) as BackyardTheme['context']['shape']

/**
 * Split the shadows in a given string
 *
 * @example
 * ```
 * splitShadows('0px 2px 1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)')
 *  => [
 *      '0px 2px 1px rgba(0, 0, 0, 0.2)',
 *      '0px 1px 1px rgba(0, 0, 0, 0.14)',
 *      '0px 1px 3px rgba(0, 0, 0, 0.12)'
 *  ]
 * ```
 *
 * @param shadows - shadow string from backyard
 * @param defaultShadow - default shadow if none given
 * @returns
 */
export const splitShadows = (
    shadows: ValueOf<BackyardTheme['shadows']>,
    defaultShadow: ValueOf<BackyardTheme['shadows']>,
) =>
    (shadows || defaultShadow).match(/[^ ,][0-9px ]+ rgba\([0-9,. ]+\)/g) as ValueOf<
        BackyardTheme['shadows']
    >[]

/**
 * Converts box-shadow to drop-shadow, if necessary.
 *
 * Use for `filter` css instead of `box-shadow`.
 *
 * @example
 * ```
 * dropShadow(theme.shadows.shadow_04)
 *  => 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.2)) 
 *          drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.14)) 
 *          drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.12))'
 * ```
 * @param shadow - box-shadow
 * @param defaultShadow - default shadow if none given
 * @returns
 */
export const dropShadow = (
    shadow: ValueOf<BackyardTheme['shadows']>,
    defaultShadow: ValueOf<BackyardTheme['shadows']> = 'none',
) =>
    // Split shadows from `box-shadow` input
    splitShadows(shadow, defaultShadow)
        // Create separate drop shadow for CSS filter
        .map((shadow) => `drop-shadow(${shadow})`)
        // Join with space
        .join(' ')
