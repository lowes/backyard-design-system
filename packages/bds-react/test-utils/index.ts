// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
