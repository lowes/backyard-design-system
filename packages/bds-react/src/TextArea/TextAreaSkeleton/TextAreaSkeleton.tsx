import * as React from 'react'
import classNames from 'classnames'

import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'
import Skeleton from '../../Skeleton'
import type { SkeletonProps, SkeletonRef } from '../../Skeleton'
import { useBackyardTheme, getShape } from '../../ThemeProvider'
import type { TextAreaProps } from '../TextArea'

/**
 * Backyard React Text Area Skeleton
 */
const TextAreaSkeleton: BDSForwardRef<TextAreaSkeletonProps> = React.forwardRef<
    TextAreaSkeletonRef,
    TextAreaSkeletonProps
>(function TextAreaSkeleton(
    {
        className,
        shape: shapeProp, // = 'rounded',
        animate = false,
        rows = 4,
        helperText,
        max,
        ...props
    },
    ref,
) {
    // Get Backyard Theme Context
    const theme = useBackyardTheme({
        validate: true,
        name: 'TextAreaSkeleton',
    })

    const { shape: shapeContext } = theme.context

    // Calculate shape
    const shape = getShape(shapeProp, shapeContext)

    return (
        <>
            <Skeleton
                className={classNames('textarea--skeleton', className)}
                variant="rect"
                shape={shape}
                animate={animate}
                height={`calc(${rows > 1 ? rows + 1 : '0'}.5rem + ${theme.sizes.size_32})`}
                {...props}
                ref={ref}
            />
            {max || helperText ? (
                <div
                    style={{
                        display: 'flex',
                        marginTop: theme.sizes.size_8,
                        padding: `0 ${theme.sizes.size_16}`,
                    }}
                >
                    {helperText ? (
                        <Skeleton
                            className="helper-text--skeleton"
                            variant="text"
                            shape={shape}
                            animate={animate}
                            style={{
                                width: '100%',
                                float: 'left',
                            }}
                        />
                    ) : (
                        <div style={{ width: '100%' }} />
                    )}
                    {max ? (
                        <Skeleton
                            className="helper-counter--skeleton"
                            variant="text"
                            shape={shape}
                            animate={animate}
                            style={{
                                width: theme.sizes.size_44,
                                marginLeft: helperText ? theme.sizes.size_16 : 0,
                                float: 'right',
                            }}
                        />
                    ) : null}
                </div>
            ) : null}
        </>
    )
})

type TextAreaSkeletonRef = SkeletonRef

type TextAreaSkeletonOverrideProps = 'max'

interface TextAreaSkeletonProps extends Omit<SkeletonProps, TextAreaSkeletonOverrideProps> {
    /**
     * Rows of the text area
     */
    rows?: TextAreaProps['rows']
    /**
     * Whether or not textarea has `helperText` prop
     */
    helperText?: boolean
    /**
     * Whether or not textarea has `max` prop
     */
    max?: boolean
}

TextAreaSkeleton.bdsName = 'TextAreaSkeleton'

export { TextAreaSkeleton }

export type { TextAreaSkeletonProps, TextAreaSkeletonRef }

export default TextAreaSkeleton
