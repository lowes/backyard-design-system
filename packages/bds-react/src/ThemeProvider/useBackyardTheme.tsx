import * as React from 'react'

import { ThemeContext } from './ThemeProvider'
import type { BackyardTheme } from './ThemeProvider'

const validateBackyardTheme = (theme: BackyardTheme, name: BackyardTheme['name']) => {
    if (!theme?.name) {
        // eslint-disable-next-line no-console
        throw Error(
            `Backyard: A Backyard component ${
                name ? `(\`${name}\`)` : ''
            } is missing a Backyard Theme.\n` +
                'This component uses Backyard tokens and a Backyard Theme is required.\n' +
                `Please wrap ${
                    name ? `\`${name}\` or` : ''
                } your entire application in a Backyard \`ThemeProvider\`.\n` +
                'For a getting started guide, see \n' +
                '`https://backyard.lowes.com/GetStarted/Develop#create-theme-provider`',
        )
    }

    return true
}

const useBackyardTheme = (props: UseBackyardThemeProps = {}) => {
    const { validate = false, name }: UseBackyardThemeProps = props

    const theme = React.useContext<BackyardTheme>(ThemeContext)

    if (validate && process.env.NODE_ENV !== 'production') {
        React.useEffect(() => {
            validateBackyardTheme(theme, name)
        }, [theme])
    }

    return theme
}

interface UseBackyardThemeProps {
    /**
     * Whether or not to validate component
     */
    validate?: boolean
    /**
     * Name of the component to validate
     */
    name?: string
}

export { useBackyardTheme, validateBackyardTheme }

export default useBackyardTheme
