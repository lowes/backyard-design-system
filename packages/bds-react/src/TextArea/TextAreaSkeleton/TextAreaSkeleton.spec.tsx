import * as React from 'react'
import { render, act } from '../../../test-utils'

import { ThemeProvider } from '../../ThemeProvider'

import TextAreaSkeleton from './TextAreaSkeleton'
import type { TextAreaSkeletonProps } from './TextAreaSkeleton'

describe('TextAreaSkeleton Snapshots', () => {
    test('renders', async () => {
        let wrapper

        await act(async () => {
            wrapper = render(
                <ThemeProvider>
                    <TextAreaSkeleton animate />
                </ThemeProvider>,
            )
        })

        const { asFragment } = wrapper

        expect(asFragment()).toMatchSnapshot()
    })

    test('renders with helper text', async () => {
        let wrapper

        await act(async () => {
            wrapper = render(
                <ThemeProvider>
                    <TextAreaSkeleton animate helperText />
                </ThemeProvider>,
            )
        })

        const { asFragment } = wrapper

        expect(asFragment()).toMatchSnapshot()
    })

    test('renders with max', async () => {
        let wrapper

        await act(async () => {
            wrapper = render(
                <ThemeProvider>
                    <TextAreaSkeleton animate max />
                </ThemeProvider>,
            )
        })

        const { asFragment } = wrapper

        expect(asFragment()).toMatchSnapshot()
    })

    test('renders with max and helper text', async () => {
        let wrapper

        await act(async () => {
            wrapper = render(
                <ThemeProvider>
                    <TextAreaSkeleton animate max helperText />
                </ThemeProvider>,
            )
        })

        const { asFragment } = wrapper

        expect(asFragment()).toMatchSnapshot()
    })

    test('rows snapshots', () => {
        const rows: TextAreaSkeletonProps['rows'][] = [4, 8, 12]

        rows.forEach((row) => {
            const { asFragment } = render(<TextAreaSkeleton rows={row} />)

            expect(asFragment()).toMatchSnapshot()
        })
    })
})
