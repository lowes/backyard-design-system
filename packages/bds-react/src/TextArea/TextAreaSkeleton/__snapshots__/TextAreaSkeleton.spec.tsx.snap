// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaSkeleton Snapshots renders 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(5.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(5.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--animated"
    height="calc(5.5rem + 2rem)"
  />
</DocumentFragment>
`;

exports[`TextAreaSkeleton Snapshots renders with helper text 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(5.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(5.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--animated"
    height="calc(5.5rem + 2rem)"
  />
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    style="display: flex; margin-top: 0.5rem; padding: 0px 1rem;"
  >
    <div
      class="c0 variant--text shape--rounded helper-text--skeleton style--animated"
      style="width: 100%; float: left;"
    />
  </div>
</DocumentFragment>
`;

exports[`TextAreaSkeleton Snapshots renders with max 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(5.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(5.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--animated"
    height="calc(5.5rem + 2rem)"
  />
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    style="display: flex; margin-top: 0.5rem; padding: 0px 1rem;"
  >
    <div
      style="width: 100%;"
    />
    <div
      class="c0 variant--text shape--rounded helper-counter--skeleton style--animated"
      style="width: 2.75rem; margin-left: 0px; float: right;"
    />
  </div>
</DocumentFragment>
`;

exports[`TextAreaSkeleton Snapshots renders with max and helper text 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(5.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(5.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--animated"
    height="calc(5.5rem + 2rem)"
  />
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    style="display: flex; margin-top: 0.5rem; padding: 0px 1rem;"
  >
    <div
      class="c0 variant--text shape--rounded helper-text--skeleton style--animated"
      style="width: 100%; float: left;"
    />
    <div
      class="c0 variant--text shape--rounded helper-counter--skeleton style--animated"
      style="width: 2.75rem; margin-left: 1rem; float: right;"
    />
  </div>
</DocumentFragment>
`;

exports[`TextAreaSkeleton Snapshots rows snapshots 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(5.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(5.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--default"
    height="calc(5.5rem + 2rem)"
  />
</DocumentFragment>
`;

exports[`TextAreaSkeleton Snapshots rows snapshots 2`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(9.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(9.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--default"
    height="calc(9.5rem + 2rem)"
  />
</DocumentFragment>
`;

exports[`TextAreaSkeleton Snapshots rows snapshots 3`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  height: calc(13.5rem + 2rem);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  height: calc(13.5rem + 2rem);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--rect shape--rounded textarea--skeleton style--default"
    height="calc(13.5rem + 2rem)"
  />
</DocumentFragment>
`;
